builds/azure-pipelines/build-release.yml

```diff
@@ -18,7 +27,7 @@ variables:
   versionMajorMinor: '$(versionMajor).$(versionMinor)'  # This variable is only used for the counter so we reset properly when either major or minor is bumped
   versionPatch: $[counter(variables['versionMajorMinor'], 0)] # This will reset when we bump minor version
   binariesVersion: '$(versionMajor).$(versionMinor).$(versionPatch)'
-  nugetVersion: '$(binariesVersion)-preview'
+  nugetVersion: $[replace(replace('True', eq(parameters.ReleaseType, 'full'), '$(binariesVersion)'),'True','$(binariesVersion)-preview')] # This will set the preview tag to the version based on the ReleaseType parameter.
```

Comment:
There is no ternary operator in ADO expressions, so used the [replace](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops#replace) as a hack for similar effect.